### PR TITLE
Run dapps tests via reusable workflow

### PR DESCRIPTION
**Pull request by lyubenkov**

### CODE REVIEW

**Comment by lyubenkov** [comment]
Green build https://github.com/neonlabsorg/neon-tests/actions/runs/8648882153

**Comment by lyubenkov** [comment]
@kristinaNikolaevaa please verify, in case there're no suggestions I'll change the branch in the workflow to `develop`.

File: .github/workflows/dapps.yml [hunk 1]
```diff
@@ -40,8 +40,51 @@ on:
         type: string
         required: false
         description: "Url to send the report as comment for PR"
+  workflow_call:
+    inputs:
+        network:
+          type: string
+          default: terraform
```
Comment (kristinaNikolaevaa) [inline .github/workflows/dapps.yml]: custom?

File: .github/workflows/dapps.yml [hunk 2]
```diff
@@ -88,21 +135,21 @@ jobs:
         with:
           network: ${{ env.NETWORK }}
           devnet_solana_url: ${{ secrets.SOLANA_URL }}
-          proxy_ip: ${{ steps.prepare_terraform.outputs.proxy_ip }}
-          solana_ip: ${{ steps.prepare_terraform.outputs.solana_ip }}
-          proxy_url: ${{ github.event.inputs.proxy_url }}
-          solana_url: ${{ github.event.inputs.solana_url }}
-          faucet_url: ${{ github.event.inputs.faucet_url }}
-          network_id: ${{ github.event.inputs.network_id }}
+          proxy_ip: ${{ steps.prepare_terraform.outputs.proxy_ip || inputs.proxy_ip }}
```
Comment (kristinaNikolaevaa) [inline .github/workflows/dapps.yml]: I think It is unnecessary, steps.prepare_terraform.outputs always returns something

File: .github/workflows/dapps.yml [hunk 3]
```diff
@@ -88,21 +135,21 @@ jobs:
...
     outputs:
       network: ${{ env.NETWORK }}
       solana_url: ${{ steps.vars.outputs.solana_url }}
       proxy_url: ${{ steps.vars.outputs.proxy_url }}
       faucet_url:  ${{ steps.vars.outputs.faucet_url }}
       network_id: ${{ steps.vars.outputs.network_id }}
-      proxy_ip: ${{ steps.prepare_terraform.outputs.proxy_ip }}
-      solana_ip: ${{ steps.prepare_terraform.outputs.solana_ip }}
+      proxy_ip: ${{ steps.prepare_terraform.outputs.proxy_ip || inputs.proxy_ip }}
+      solana_ip: ${{ steps.prepare_terraform.outputs.solana_ip || inputs.solana_ip }}
```
Comment (kristinaNikolaevaa) [inline .github/workflows/dapps.yml]: I think It is unnecessary, steps.prepare_terraform.outputs always returns something

File: .github/workflows/dapps.yml (hunk 1 above)
Comment (lyubenkov) [reply .github/workflows/dapps.yml]: Fixed

File: .github/workflows/dapps.yml (hunk 2 above)
Comment (lyubenkov) [reply .github/workflows/dapps.yml]: It was necessary when it was called outside from proxy repo, but now, after splitting the flow, it's not needed. Removed.

File: .github/workflows/dapps.yml (hunk 3 above)
Comment (lyubenkov) [reply .github/workflows/dapps.yml]: It was necessary when it was called outside from proxy repo, but now, after splitting the flow, it's not needed. Removed.

File: .github/workflows/dapps.yml [hunk 4]
```diff
@@ -105,525 +105,25 @@ jobs:
...
-          name: pancake-report
-          path: reports/
-      - name: "Swap report"
-        run: |
-          python3 ./clickfile.py dapps report --directory=reports \
-          --pr_url_for_report=${{ github.event.inputs.pr_url_for_report }} \
-          --token=${{secrets.GHTOKEN}}
+    uses: ./.github/workflows/dapps_reusable.yml
+    with:
+      network: custom
```
Comment (kristinaNikolaevaa) [inline .github/workflows/dapps.yml]: terraform for pipeline in neon-tests
custom for pipeline in proxy

File: .github/workflows/dapps_reusable.yml [hunk 5]
```diff
@@ -0,0 +1,602 @@
+name: dApps testing
+run-name: dApps testing (${{ inputs.dapps || 'all dapps'}}, ${{ inputs.network }})
```
Comment (kristinaNikolaevaa) [inline .github/workflows/dapps_reusable.yml]: Lets rename 
now there are 2 the same names
My suggesion
1)dApps tests
2)dApps jobs for external stand

**Review by kristinaNikolaevaa — APPROVED** [review]

File: .github/workflows/dapps_reusable.yml (hunk 5 above)
Comment (lyubenkov) [reply .github/workflows/dapps_reusable.yml]: Agreed to use dApps reusable, as dApps tests also use dApps jobs for external stand inside.

File: .github/workflows/dapps.yml (hunk 4 above)
Comment (lyubenkov) [reply .github/workflows/dapps.yml]: Changed to terraform

**Pull request merged**